### PR TITLE
feat: Support for complex PHPDoc types in `fully_qualified_strict_types`

### LINKED ISSUE #7619: Support for complex PHPDoc types in `fully_qualified_strict_types`
## Feature request

With the latest release - v3.42.0 - the `fully_qualified_strict_types` rule does fix some phpdoc to imported types, but not all.

The following code:

```php
<?php

use DataTables;
use DataTables\Database;
use DataTables\Editor;
use DataTables\SearchPaneOptions;

/**
 * @param SearchPaneOptions|callable(Database, \DataTables\Editor): (array|bool) $spInput
 */
function searchPaneOptions($spInput = null) {}
```

is fixed to:
```php
<?php

use DataTables\Database;
use DataTables\SearchPaneOptions;

/**
 * @param SearchPaneOptions|callable(Database, \DataTables\Editor): (array|bool) $spInput
 */
function searchPaneOptions($spInput = null) {}
```
applied rules: braces_position, no_unused_imports, single_line_empty_body

but I would expect the following result instead:
```php
<?php

use DataTables\Database;
use DataTables\Editor;
use DataTables\SearchPaneOptions;

/**
 * @param SearchPaneOptions|callable(Database, Editor): (array|bool) $spInput
 */
function searchPaneOptions($spInput = null) {}
```

### PR DESCRIPTION
**Pull request by mvorisek**
closes #7619 🔥

needs #8076, #8077, #8106 PRs

### CODE REVIEW

File: src/Fixer/Import/FullyQualifiedStrictTypesFixer.php [hunk 1]
```diff
@@ -649,34 +649,47 @@ private function fixPhpDoc(Tokens $tokens, int $index, array $uses, string $name
...
+     * @param _Uses $uses
+     */
+    private function fixPhpDocType(string $type, array $uses, string $namespaceName): string
+    {
+        $typeExpression = new TypeExpression($type, null, []);
+
+        $typeExpression = $typeExpression->mapTypes(function (TypeExpression $type) use ($uses, $namespaceName) {
+            $v = $type->toString();
+
+            if ($type->isUnionType() || !Preg::match('/^'.self::REGEX_CLASS.'$/', $v)) {
```
Comment (Wirone) [inline src/Fixer/Import/FullyQualifiedStrictTypesFixer.php:670]: After #8214 most probably you mean this here:

```suggestion
            if ($type->isCompositeType() || !Preg::match('/^'.self::REGEX_CLASS.'$/', $v)) {
```

**Review by Wirone — CHANGES_REQUESTED** [review]
Maybe it would be good to add more test cases, but I don't have enough time to validate if all the complex cases are covered now. Looks very promising anyway, let's address my concerns for now.

File: src/Fixer/Import/FullyQualifiedStrictTypesFixer.php (hunk 1 above)
Comment (mvorisek) [reply src/Fixer/Import/FullyQualifiedStrictTypesFixer.php]: done

**Comment by mvorisek** [comment]
PR is done. There are only a few changed lines when not counting the moved code to a private method - all hard challenges has been landed before. With this said, the added/updated tests are covering this PR well.

**Review by Wirone — APPROVED** [review]

**Pull request merged**